### PR TITLE
[t5] analytic feature

### PR DESCRIPTION
**Pull request by KenXiong123**
image
image
1. Finance User will see loan status and requester name; Personal user will see null
2. All user have expenditure break down as report

TODO:
1. Tests need to be written in the next pr

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/4156db
 spring.datasource.username=root
-spring.datasource.password=dbuserdbuser
+spring.datasource.password=Xkqwert0301!
```
Comment (howieraem) [inline src/main/resources/application.properties:3]: pls don't change this

File: src/test/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/4156dbtest
 spring.datasource.username=root
-spring.datasource.password=dbuserdbuser
+spring.datasource.password=Xkqwert0301!
```
Comment (howieraem) [inline src/test/resources/application.properties:3]: pls don't change this

File: src/main/java/com/lgtm/easymoney/controllers/AnalyticController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  @Autowired
+  public AnalyticController(AnalyticService analyticService) {
+    this.analyticService = analyticService;
+  }
+
+  @GetMapping("/{id}")
+  @Operation(summary = "Method to retrieve the analytic report of a user by user ID.")
+  public ResponseEntity<AnalyticRsp> getAnalytic(@PathVariable(value = "id") @NotNull Long id) {
+    return new ResponseEntity<>(analyticService.getAnalytic(id), HttpStatus.OK);
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/controllers/AnalyticController.java:32]: Can a user view other's analytics (even if he/she logged in)? It doesn't sound right. You should follow the @CurrentUser pattern in other controllers

File: src/main/java/com/lgtm/easymoney/services/AnalyticService.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+
+import com.lgtm.easymoney.models.User;
+import com.lgtm.easymoney.payload.rsp.AnalyticRsp;
+
+/**
+ * Analytic Service Interface.
+ */
+public interface AnalyticService {
+  AnalyticRsp getAnalytic(Long uid);
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/services/AnalyticService.java:11]: In @CurrentUser pattern, the input should be a User object instead

**Review by howieraem — CHANGES_REQUESTED** [review]

**Comment by rl3250** [comment]
why GH actions isn't running on this PR? hmm

File: src/main/java/com/lgtm/easymoney/controllers/AnalyticController.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@RequestMapping("/analytic")
+public class AnalyticController {
+  private final AnalyticService analyticService;
+
+  @Autowired
+  public AnalyticController(AnalyticService analyticService) {
+    this.analyticService = analyticService;
+  }
+
+  @GetMapping
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/controllers/AnalyticController.java:32]: Please add the swagger security annotation here (see how other controllers are written)

**Review by howieraem — CHANGES_REQUESTED** [review]

**Review by howieraem — APPROVED** [review]

**Pull request merged**